esp32_s3_eye/include/bsp/esp32_s3_eye.h

```diff
@@ -0,0 +1,289 @@
+
+/**************************************************************************************************
+ *
+ * I2C interface
+ *
+ * There are two devices connected to I2C peripheral:
+ *  - QMA7981 Inertial measurement unit
+ *  - OV2640 Camera module
+ **************************************************************************************************/
+#define BSP_I2C_NUM     CONFIG_BSP_I2C_NUM
+
+/**
+ * @brief Init I2C driver
+ *
+ */
+void bsp_i2c_init(void);
+
+/**
+ * @brief Deinit I2C driver and free its resources
+ *
+ */
+void bsp_i2c_deinit(void);
+
+/**************************************************************************************************
+ *
+ * Camera interface
+ *
+ * ESP32-S3-EYE is shipped with OV2640 camera module.
+ * As a camera driver, esp32-camera component is used.
+ *
+ * Example configuration:
+ * \code{.c}
+ * const camera_config_t camera_config = BSP_CAMERA_DEFAULT_CONFIG;
+ * esp_err_t err = esp_camera_init(&camera_config);
+ * \endcode
+ **************************************************************************************************/
+/**
+ * @brief ESP32-S3-EYE camera default configuration
+ *
+ * In default configuration we select RGB565 color format and 240x240 image size
+ */
+#define BSP_CAMERA_DEFAULT_CONFIG         \
+    {                                     \
+        .pin_pwdn = GPIO_NUM_NC,          \
+        .pin_reset = GPIO_NUM_NC,         \
+        .pin_xclk = BSP_CAMERA_XCLK,      \
+        .pin_sscb_sda = GPIO_NUM_NC,      \
+        .pin_sscb_scl = GPIO_NUM_NC,      \
+        .pin_d7 = BSP_CAMERA_D7,          \
+        .pin_d6 = BSP_CAMERA_D6,          \
+        .pin_d5 = BSP_CAMERA_D5,          \
+        .pin_d4 = BSP_CAMERA_D4,          \
+        .pin_d3 = BSP_CAMERA_D3,          \
+        .pin_d2 = BSP_CAMERA_D2,          \
+        .pin_d1 = BSP_CAMERA_D1,          \
+        .pin_d0 = BSP_CAMERA_D0,          \
+        .pin_vsync = BSP_CAMERA_VSYNC,    \
+        .pin_href = BSP_CAMERA_HSYNC,     \
+        .pin_pclk = BSP_CAMERA_PCLK,      \
+        .xclk_freq_hz = 16000000,         \
+        .ledc_timer = LEDC_TIMER_0,       \
+        .ledc_channel = LEDC_CHANNEL_0,   \
+        .pixel_format = PIXFORMAT_RGB565, \
+        .frame_size = FRAMESIZE_240X240,  \
+        .jpeg_quality = 12,               \
+        .fb_count = 2,                    \
+        .fb_location = CAMERA_FB_IN_PSRAM \
+    }
+
+/**************************************************************************************************
+ *
+ * uSD card
+ *
+ * After mounting the uSD card, it can be accessed with stdio functions ie.:
+ * \code{.c}
+ * FILE* f = fopen(BSP_MOUNT_POINT"/hello.txt", "w");
+ * fprintf(f, "Hello %s!\n", bsp_sdcard->cid.name);
+ * fclose(f);
+ * \endcode
+ **************************************************************************************************/
+#define BSP_MOUNT_POINT      CONFIG_BSP_SD_MOUNT_POINT
+extern sdmmc_card_t *bsp_sdcard;
```

Comment:
Why the `sdmmc_card_t` is not returned from `bsp_sdcard_mount`? Must it be the `*bsp_sdcard` global?

Reply:
I agree that the API is far from optimal. But at least, ATM, we have the same API for all boards. I'll create a ticket to update the API for all boards